data_pipeline/twitter/data_from_twitter_streaming_api.py

```diff
@@ -0,0 +1,129 @@
+    """ A listener handles tweets that are received from the stream.
+    """
+
+    def __init__(
+            self, streaming_config: TwitterDataPipelineConfig,
+            multi_processing: bool = None, num_workers: int = None
+    ):
+        super().__init__()
+        self.client = Client()
+        self.streaming_config = streaming_config
+        self.bq_table = get_bq_table(
+            streaming_config.gcp_project,
+            streaming_config.dataset,
+            streaming_config.tweet_table
+        )
+        queue, worker_func = get_queue_and_worker_func(multi_processing)
+        self.queue = queue()
+        num_workers = num_workers or 1
+        self._spin_worker(num_workers, worker_func)
+        self.filter_regex_pattern = re.compile(
+            '|'.join(streaming_config.etl_terms).lower())
+        self.tweet_counter = 0
+
+    def _spin_worker(self, num_workers, worker_func):
+        for _ in range(num_workers):
+            worker = worker_func(target=self.process_queue)
+            worker.daemon = True
+            worker.start()
+
+    # pylint: disable=(arguments-differ
+    def on_data(self, data):
+        self.tweet_counter += 1
+        if self.tweet_counter % 50 == 0:
+            LOGGER.info("Matched %d so far", self.tweet_counter)
+        self.queue.put(data)
+
+    def process_queue(self):
+        while True:
+            self.process_data(self.queue.get())
+            self.queue.task_done()
+
+    def process_data(self, data):
+        tweets_in_reponse = []
+        current_timestamp_as_provenance = get_current_timestamp_as_string(
+            ETL_TIMESTAMP_FORMAT
+        )
+
+        record = json.load(StringIO(data))
+        while True:
+            extracted_tweet = extract_tweet_from_twitter_response(record)
+            if self.filter_regex_pattern.search(extracted_tweet.get(
+                    TwitterPipelineModuleConstants.TWEET_TEXT, ''
+            ).lower()):
+                tweets_in_reponse.append(
+                    add_provenance_to_record(
+                        extracted_tweet, current_timestamp_as_provenance,
+                        self.streaming_config.timestamp_field_name,
+                        self.streaming_config.provenance_field_name,
+                        annotation_field_name='annotation',
+                        record_annotation={
+                            'keywords':
+                                self.streaming_config.etl_terms
+                        }
+                    )
+                )
+            record = (
+                record.get('retweeted_status')
+                or record.get('quoted_status')
+            )
+            if not record:
+                break
+        if len(tweets_in_reponse) > 0:
+            self.client.insert_rows_json(
+                table=self.bq_table, json_rows=tweets_in_reponse
+            )
+
+    def on_error(self, status):
+        LOGGER.debug(status)
+
+
+def etl_track_keywords(
```

Comment:
Is this streaming etl already used? Or will you add it as part of this PR or another?